Tasks/VsTestPlatformToolInstaller/vstestplatformtoolinstaller.ts

```diff
@@ -5,34 +5,45 @@ import * as restm from 'typed-rest-client/RestClient';
 import {IExecOptions, IExecSyncResult, ToolRunner} from "vsts-task-lib/toolrunner";
 import * as os from 'os';
 import * as path from 'path';
+import * as ci from './cieventlogger';
+let perf = require('performance-now');
 
+let executionStartTime = perf();
 let osPlat: string = os.platform();
 let packageName = 'Microsoft.TestPlatform';
 let packageSource = 'https://api.nuget.org/v3/index.json'
 
 async function startInstaller() {
+    tl.setResourcePath(path.join(__dirname, 'task.json'));
+    ci.publishEvent('Start', { OS: osPlat, isSupportedOS: (osPlat === 'win32').toString(), startTime: executionStartTime } );
+
     if (osPlat !== 'win32') {
-        throw new Error(tl.loc('OnlyWindowsOsSupported'));
+        // Fail the task if os is not windows
+        taskLib.setResult(taskLib.TaskResult.Failed, tl.loc('OnlyWindowsOsSupported'));
+        return;
     }
 
     try {
-        tl.setResourcePath(path.join(__dirname, 'task.json'));
-
         console.log(tl.loc('StartingInstaller'));
         console.log('==============================================================================');
 
         // Read inputs
         let versionSelectorInput = taskLib.getInput('versionSelector', true);
-        let testPlatformVersion = taskLib.getInput('testPlatformVersion', false); 
+        let testPlatformVersion = taskLib.getInput('testPlatformVersion', false);
+
+        ci.publishEvent('Options', { versionSelectorInput: versionSelectorInput, testPlatformVersion: testPlatformVersion } );
 
         //TODO: Add an input for cleaning up the tool cache?
         
         // Get the required version of the platform and make necessary preparation to allow its consumption down the phase 
         await getVsTestPlatformTool(testPlatformVersion, versionSelectorInput);
     }
     catch (error) {
+        ci.publishEvent('Completed', { isSetupSuccessful: 'false', error: error.message } );
```

Comment:
can we have additional properties indicating that it is from the task installer..?

Reply:
There is a dedicated feature for tools installer